Week3/homework/step2-6.js

```diff
@@ -4,13 +4,24 @@ const arr2d = [[1, 2], [3, 4], [5, 6]];
 const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
 
 function flattenArray2d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  const result = arr.reduce((total, item) => total.concat(item));
+  return result;
 }
 
-function flattenArray3d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+function flattenArray3d(arr, result = []) {
+  // const result = arr;
+  // for (let i = 0; i < 2; i++) {
+  //   result.reduce((accu, curr) => accu.concat(curr));
+  // }
+  arr.forEach(i => {
+    if (Array.isArray(i)) {
+      flattenArray3d(i, result);
+      // if it is _K_ dimensions we call i k times .
+    } else {
+      result.push(i);
+    }
+  });
+  return result;
 }
```

Comment:
Nice solution, using a recursive function with a default parameter. It will work with arrays of any dimension. 👍